### PR TITLE
IOS-9951: Make changes to all BOXRequest: GET request only calls API …

### PR DESCRIPTION
**Pull request by thuyBox**
…and update cache if valid refreshBlock, for non-GET requests, always call API and update cache regardless of refreshBlock

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @thuyBox, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by thuyBox** [comment]
@scottliubox @adempsey @sowmiya

File: BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkCommentsRequest.h [hunk 1]
```diff
@@ -10,7 +10,11 @@
 @property (nonatomic, readonly, strong) NSString *bookmarkID;
 
 - (instancetype)initWithBookmarkID:(NSString *)bookmarkID;
+
+//Perform API request and any cache update only if completionBlock is not nil
 - (void)performRequestWithCompletion:(BOXObjectsArrayCompletionBlock)completionBlock;
+
+//Perform API request and any cache update only if completionBlock is not nil
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkCommentsRequest.h]: refreshBlock

File: BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkRequest.h [hunk 2]
```diff
@@ -12,7 +12,11 @@
 @property (nonatomic, readonly, strong) NSString *bookmarkID;
 
 - (instancetype)initWithBookmarkID:(NSString *)bookmarkID;
+
+//Perform API request and any cache update only if completionBlock is not nil
 - (void)performRequestWithCompletion:(BOXBookmarkBlock)completionBlock;
+
+//Perform API request and any cache update only if completionBlock is not nil
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkRequest.h]: refresh

File: BoxContentSDK/BoxContentSDK/Requests/BOXCollaborationPendingRequest.m [hunk 3]
```diff
@@ -62,9 +62,8 @@ - (void)performRequestWithCompletion:(BOXCollaborationArrayCompletionBlock)compl
                 completionBlock(nil, error);
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXCollaborationPendingRequest.m:62]: Move the "BOOL isMainThread ..." into the if block?

File: BoxContentSDK/BoxContentSDK/Requests/BOXMetadataRequest.m [hunk 4]
```diff
@@ -94,9 +94,8 @@ - (void)performRequestWithCompletion:(BOXMetadatasBlock)completionBlock
                 completionBlock(nil, error);
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXMetadataRequest.m:94]: Move the "BOOL isMainThread" into the if block?

File: BoxContentSDK/BoxContentSDK/Requests/BOXMetadataTemplateRequest.m [hunk 5]
```diff
@@ -94,9 +94,8 @@ - (void)performRequestWithCompletion:(BOXMetadataTemplatesBlock)completionBlock
                 completionBlock(nil, error);
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXMetadataTemplateRequest.m:94]: Move those precursor local variables into the if block?

File: BoxContentSDK/BoxContentSDK/Requests/BOXSearchRequest.m [hunk 6]
```diff
@@ -155,9 +155,8 @@ - (void)performRequestWithCompletion:(BOXItemArrayCompletionBlock)completionBloc
                 completionBlock(nil, 0, NSMakeRange(0, 0), error);
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXSearchRequest.m:155]: Move local vars into if block?

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedItemArrayRequest.m [hunk 7]
```diff
@@ -84,9 +84,8 @@ - (void)performRequestWithCompletion:(BOXItemArrayCompletionBlock)completionBloc
                 completionBlock(nil, 0, NSMakeRange(0, 0), error);
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXTrashedItemArrayRequest.m:84]: move precursor local vars into if block?

**Comment by boxtcarpel** [comment]
+2 LGTM

:izakaya_lantern:

File: BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkCommentsRequest.h (hunk 1 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkCommentsRequest.h]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkRequest.h (hunk 2 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXBookmarkRequest.h]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXCollaborationPendingRequest.m (hunk 3 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXCollaborationPendingRequest.m:62]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXMetadataRequest.m (hunk 4 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXMetadataRequest.m:94]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXMetadataTemplateRequest.m (hunk 5 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXMetadataTemplateRequest.m:94]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXSearchRequest.m (hunk 6 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXSearchRequest.m:155]: done

**Pull request merged**